File: pset1/greedy/greedy.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n, count;

    // Asks the user how much change is owed.
    do
    {
        n = get_float("Enter dollar amount:") * 100;
    }
    while (n < 0);

    // Count the minimum number of coins.
    count = n / 25;
    n %= 25;

    count += n / 10;
    n %= 10;

    count += n / 5;
    n %= 5;

    count += n;

    // Show the result.
    printf("%i\n", count);
}

File: pset6/sentimental/greedy/greedy.py
# Count the minimum number of coins

from cs50 import get_float
from math import floor

def main():
    # Get user input
    cash = floor(100 * get_positive_float("How much money do you need to transfer? "))

    # Determine minimal number of coins
    coins = cash // 25
    cash %= 25

    coins += cash // 10
    cash %= 10

    coins += cash // 5
    cash %= 5

    coins += cash

    # Print the result
    print(coins)

def get_positive_float(message):
    """Ask the user until the number is positive"""
    while True:
        f = get_float(message)
        if (f > 0):
            return f;

if __name__ == "__main__":
    main()